core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -226,20 +229,26 @@ class ReplicaFetcherThread(name: String,
     }
   }
 
-  override def buildFetchRequest(partitionMap: Seq[(TopicPartition, PartitionFetchState)]): FetchRequest = {
+  override def buildFetchRequest(partitionMap: Seq[(TopicPartition, PartitionFetchState)]): ResultWithPartitions[FetchRequest] = {
     val requestMap = new util.LinkedHashMap[TopicPartition, JFetchRequest.PartitionData]
+    val partitionsWithError = mutable.Set[TopicPartition]()
 
     partitionMap.foreach { case (topicPartition, partitionFetchState) =>
       // We will not include a replica in the fetch request if it should be throttled.
       if (partitionFetchState.isReadyForFetch && !shouldFollowerThrottle(quota, topicPartition)) {
-        val logStartOffset = replicaMgr.getReplicaOrException(topicPartition).logStartOffset
-        requestMap.put(topicPartition, new JFetchRequest.PartitionData(partitionFetchState.fetchOffset, logStartOffset, fetchSize))
+        try {
+          val logStartOffset = replicaMgr.getReplicaOrException(topicPartition).logStartOffset
+          requestMap.put(topicPartition, new JFetchRequest.PartitionData(partitionFetchState.fetchOffset, logStartOffset, fetchSize))
+        } catch {
+          case e: KafkaStorageException =>
```

Comment:
Hmm, we're losing the error message here. Do we not want to log it?

Reply:
This patch does not remove any existing error message. I assume you are suggesting that we should log error if KafkaStorageException is caught here, right?

I think it is OK not to log error message if KafkaStorageException is caught. KafkaStorageException will only be caught here if the replica is marked offline in ReplicaManager.handleLogDirFailure(). In this case, there should already be error message in the log that shows this replica is marked offline as well as the cause of the original log dir failure event. Does this make sense?

Reply:
Thanks for the explanation. Would it not be worth logging at debug or trace level even? Is the concern that this could be noisy?

If you think it's not worth logging because it would be noisy, then please add a comment with your explanation and change `e` to `_` since the exception is not used.

Reply:
Sure. I will change `e` to `_` and log partitionsWithError at debug level in `handlePartitionsWithErrors`. Does this address the problem?

I am not sure whether logging this would be noisy. It is just not the focus of the patch to add additional debug level logging. The policy around what log is necessary for debug level logging is not very clear. I am usually inclined not to adding additional logging unless there is clear usage of the log. Is there specific usage of this debugging level logging? Do you have any guideline that I could follow going forward regarding when we should add debug level logging?

Reply:
I just raised it because it's generally a bad practice to catch an exception and ignore the error message. If there's a good reason to do it, we should add a comment. The code to catch and ignore was added in this PR and hence my comment.

With regards to `handlePartitionsWithErrors`, maybe we should keep the debug logging where it is as this can be very noisy when there are issues with the leader. I asked because it seemed a bit arbitrary to log just in one case. But perhaps the right solution is to log even less, so let's leave that for another PR.

Thanks for being responsive to the comments.

Reply:
Thanks for the explanation. Sure, I will not add debug level logging in handlePartitionsWithErrors(). And I will add the following comment:

```
// The replica has already been marked offline due to log directory failure and the original failure should have been logged.
// This partition will removed from ReplicaFetcherThread soon by ReplicaManager.handleLogDirFailure()
```